packages/x-utils/src/__tests__/typeguards.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { isFunction, isObject } from '../typeguards';
+
+describe('typeguards test', () => {
+  const str = 'test';
+  const num = 0;
+  const bool = false;
+  // TODO: Migrate noOp function from x-components to x-utils
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const func = (): any => {};
+  const arr = [1, 2, 3];
+  const obj = {
+    a: 1,
+    b: {
+      c: 'c'
+    }
+  };
+
+  describe('isFunction', () => {
```

Comment:
Should we test `null` and `undefined` values too?

Reply:
You're right, done!